src/nvim/api/buffer.c

```diff
@@ -702,6 +707,225 @@ ArrayOf(Integer, 2) nvim_buf_get_mark(Buffer buffer, String name, Error *err)
+    return rv;
+  }
+
+  linenr_T l_lnum;
+  colnr_T l_col;
+  bool l_set = false;
+  SET_EXTMARK_INDEX_FROM_OBJ(lower, l_lnum, l_col, l_set);
+  if (!l_set) {
+    return rv;
+  }
+
+  linenr_T u_lnum;
+  colnr_T u_col;
+  bool u_set = false;
+  SET_EXTMARK_INDEX_FROM_OBJ(upper, u_lnum, u_col, u_set);
+  if (!u_set) {
+    return rv;
+  }
+
+  ExtmarkArray *extmarks_in_range;
+  buf_T *buf = find_buffer_by_handle(buffer, err);
+
+  if (reverse) {
+    extmarks_in_range = extmark_get(buf,
+                                    (uint64_t)namespace,
+                                    l_lnum,
+                                    l_col,
+                                    u_lnum,
+                                    u_col,
+                                    (int64_t)amount,
+                                    BACKWARD);
+  }
+  // Forward
+  else {
+    extmarks_in_range = extmark_get(buf,
+                                    (uint64_t)namespace,
+                                    l_lnum,
+                                    l_col,
+                                    u_lnum,
+                                    u_col,
+                                    (int64_t)amount,
+                                    FORWARD);
+
+  }
+  // FIXME
+  if (!extmarks_in_range) {
+    ADD(rv, INTEGER_OBJ(-9));
+    return rv;
+  }
+
+  Array mark = ARRAY_DICT_INIT;
+  ExtendedMark *extmark;
+  for (size_t i = 0; i < kv_size(*extmarks_in_range); i++) {
+    mark.size = 0;
+    mark.capacity = 0;
+    mark.items = 0;
+    extmark = kv_A(*extmarks_in_range, i);
+    ADD(mark, INTEGER_OBJ((Integer)extmark->mark_id));
+    ADD(mark, INTEGER_OBJ(extmark->line->lnum));
+    ADD(mark, INTEGER_OBJ(extmark->col));
+    ADD(rv, ARRAY_OBJ(mark));
+  }
+  return rv;
+}
+
+/// Returns mark info at the given position or mark index
+///
+/// @param buffer The buffer handle
+/// @param namespace a identifier returned previously with extmark_ns_create
+/// @param id (row, col) or mark_id of mark
+/// @param[out] err Details of an error that may have occurred
+/// @return (mark_id, row, col) tuple
+ArrayOf(Object) nvim_buf_get_mark_index(Buffer buffer,
+                                        Integer namespace,
+                                        Object id,
+                                        Error *err)
+    FUNC_API_SINCE(1)
+{
+  Array rv = ARRAY_DICT_INIT;
+
+  ExtendedMark *extmark = extmark_from_id_or_pos(buffer, namespace, id, err,
+                                                 true);
+
+  if (!extmark) {
+    return rv;
+  }
+  ADD(rv, INTEGER_OBJ((Integer)extmark->mark_id));
+  ADD(rv, INTEGER_OBJ((Integer)extmark->line->lnum));
+  ADD(rv, INTEGER_OBJ((Integer)extmark->col));
+  return rv;
+}
+
+/// Returns an ordered list of mark info
+///
+/// @param buffer The buffer handle
+/// @param[out] err Details of an error that may have occurred
+/// @return [(mark_id, row, col), ...]
+ArrayOf(Object) nvim_buf_get_mark_ids(Buffer buffer,
```

Comment:
this is equivalent to ` nvim_buf_get_marks(buffer, namespace, (0,0), (-1,-1), 0, false)` right?